Homework2: Anton Stychnevsky

Stychnevsky opened the pull request:
Homework2
In task 4 I use https://www.codewars.com/

Stychnevsky committed:
Homework # 2
In task 4 I use https://www.codewars.com/

Stychnevsky committed:
Change line length from 120 to 79

Stychnevsky committed:
fixed

Stychnevsky commented:
Test flake8 failed by:
W503 line break before binary operator

But, as I understand, this rule was changed on April 2016 and it is better to break before a binary operator.:
https://github.com/python/peps/commit/c59c4376ad233a62ca4b3a6060c81368bd21e85b
https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

So, I didn't change my code

belovmd commented:
A you writing like this: 
> x = y+z
> +123

in a notebook when you are solving math equations? 

I agree that it's controversial question, but however linters should pass

belovmd commented on HW2-kata1.py:
```diff
@@ -0,0 +1,18 @@
+
+
+def move_zeros(array):
+    array_without_zero = [i for i in array if (i != 0) or (type(i) == bool)]
```
`if i or (type......`

belovmd commented on HW2-kata2.py:
```diff
@@ -0,0 +1,35 @@
+to waste. You must come up with something more clever than that. """
+
+
+def longest_slide_down(pyramid):
```
💯

belovmd commented on HW2-kata3.py:
```diff
@@ -0,0 +1,56 @@
+            previous_number = number
+            previous_pos = pos
+            direction = "uphill"
+        elif number == previous_number:
```
`for pos, number: `
`if previous< number and number >= next: peak`
wouldn't be a bit shorter?

Stychnevsky replied:
It's good idea, but there is a problem with "plateaus."
For example, [1,2,2,2,2] and [1,2,2,2,3] have no peaks.
I change code using your advice, but it needs one more loop. In spite of this
 new code looks more simply. But I think it requires more operations

belovmd replied:
It's a valid. One more idea: find all local maximums as I suggested. 
Than we can find out all local minimums  previous>= num >=next
After that we can check that after each local max would be local min lesser than it - than it would be correct peak. Othervise (max = next min) it would be just plateau

belovmd commented on HW2-kata4.py:
```diff
@@ -0,0 +1,87 @@
+"""
+
+
+def justify(text, width):
```
maybe it has sense to get a list of all fords in appropriate order `words = test.split()`
next you can take from it words that no more than width (with at least one space) and write function to form string for these words? 
it seems to me it would be shorter and clearer

belovmd commented on HW2-task3.py:
```diff
@@ -0,0 +1,16 @@
+"""
+
+for number in range(1, 101):
+    if (number % 3 == 0) & (number % 5 == 0):
```
15?

belovmd commented on HW2-task1.py:
```diff
@@ -0,0 +1,24 @@
+
+number = int(input("Enter number: \n"))
+if number:
+    len_of_number = 0
```
maybe it would be easier to revert a number with just one for loop? 
this solution looks too complex

belovmd commented on HW2-kata5.py:
```diff
@@ -0,0 +1,89 @@
+lowercase characters in the english alphabet. """
+
+
+def simplify(poly):
```
it looks too complex but currently i have now idea how to simplify it.

Stychnevsky replied:
I tried to use str.split(), but it became hard to understand how many words will be in every line, and code still looked not good

Than I looked through decisions on Codewars and found a better way to split text by lines. I used this idea, and than split words into lines.
So, I think now code became simpler.

belovmd reviewed (changes requested):
You homework looks great! 
Please take a look at these issues:

Stychnevsky committed:
fixed according to comments

HW2-task1.py – fully changed, now only one loop used
HW2-task3.py – fixed according to recommendations
HW2-kata1.py - fixed according to recommendations
HW2-kata3.py - fully changed according to recommendations
HW2-kata4.py – fixed (line break after binary operator)

Stychnevsky committed:
fixed

Stychnevsky committed:
fully changed HW2-kata2.py

Stychnevsky committed:
fixed

belovmd reviewed (approved):
10
